src/libraries/Microsoft.PowerFx.Core/Public/CheckResult.cs

```diff
@@ -190,6 +191,12 @@ public CheckResult SetExpectedReturnValue(FormulaType type, bool allowCoerceTo =
             return this;
         }
 
+        public CheckResult SetAllowedInputType(FormulaType[] allowedTypes)

```

Comment:
>CheckResult SetAllowedInputType(FormulaType[] allowedTypes) [](http://example.com/codeflow?start=15&length=59)

I'm concerned about adding this overload:
- it's close to what "allow coerce" was attempting to do. 
- the host will still need to handle the possible values.

Reply:
discussed offline, this is intended.  So we should just make this an overload of SetExpectedReturnValue, and ideally 1 chains to the other.

Reply:
Updated to use SetExpectedReturnValue